File: src/main/java/xin/vanilla/entity/config/Permissions.java
package xin.vanilla.entity.config;

import java.util.HashSet;
import java.util.Set;

/**
 * 插件角色
 */
public class Permissions {
    /**
     * 超人
     */
    public Long superOwner = 0L;

    /**
     * 主人
     */
    public Long botOwner = 0L;

    /**
     * 超管
     */
    public Set<Long> superAdmin = new HashSet<>();

    /**
     * 主管
     */
    public Set<Long> botAdmin = new HashSet<>();

    /**
     * 全局副管
     * <p>与群内单独设置的群副管权限相同</p>
     */
    public Set<Long> deputyAdmin = new HashSet<>();


}

File: src/main/java/xin/vanilla/entity/config/Permissions.kt
package xin.vanilla.entity.config

import kotlinx.serialization.Serializable

/**
 * 插件角色
 */
@Serializable
class Permissions {
    /**
     * 主人
     */
    var botOwner: Long = 0L

    /**
     * 超管
     */
    var superAdmin: Set<Long> = HashSet()

    /**
     * 主管
     */
    var botAdmin: Set<Long> = HashSet()

    /**
     * 全局副管
     * <p>与群内单独设置的群副管权限相同</p>
     */
    var deputyAdmin: Set<Long> = HashSet()

}
